### ISSUE TITLE
A newSelf-Calibrated Illumination (SCI) learning framework

### ISSUE THREAD

**Issue by gao3656888** [opencv/opencv#22636]
### Descripe the feature and motivation

We develop a newSelf-Calibrated Illumination (SCI) learning framework for fast, flexible, androbust brightening images in real-world low-light scenarios. To be specific, weestablish a cascaded illumination learning process with weight sharing tohandle this task. Considering the computational burden of the cascaded pattern,we construct the self-calibrated module which realizes the convergence betweenresults of each stage, producing the gains that only use the single basic blockfor inference (yet has not been exploited in previous works), which drasticallydiminishes computation cost. We then define the unsupervised training loss toelevate the model capability that can adapt to general scenes. We think this method can be well introduced into OpenCV because of its lightweight.

### Additional context

The source code is available at https://github.com/vis-opt-group/SCI.

**Comment by fengyuentau** [comment]
I think this feature request should go to opencv_zoo: https://github.com/opencv/opencv_zoo

**Comment by gao3656888** [comment]
> I think this feature request should go to opencv_zoo: https://github.com/opencv/opencv_zoo

I tried recoding in opencv_ zoo, but several questions have been encountered. Could you answer them:

1) The type definitions of all image tasks are detection and recognition. Since our task is image enhancement, how to define a new type.

2) I did not find the network structure of the model, but only the definition of the network. Where can I find the network process structure.

3) How do I get the .onnx file that saves the model file. Whether it can be directly converted from .pt file.

Thanks a lot!

**Comment by fengyuentau** [comment]
1. It does not matter what type of the model. Contributor should focus on implementing a demo instead of the benchmarking.
2. You need to read carefully how the inference is done for the model. Typically speaking, the `test.py` in the repo should give you enough information on pre-processing, inference and post-processing.
3. PyTorch offers `torch.onnx` module for the ONNX convertion: https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html

**Comment by gao3656888** [comment]
> 1. It does not matter what type of the model is. Contributor should focus on implementing a demo instead of the benchmarking.
> 2. You need to read carefully how the inference is done for the model. Typically speaking, the `test.py` in the repo should give you enough information on pre-processing, inference and post-processing.
> 3. PyTorch offers `torch.onnx` module for the ONNX convertion: https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html

Thank you for your answer.

I have completed all aspects of the deployment.

However, we found that we could not directly call cv.FactorDetectorYN like the introduced YuNet, so we could not use benchmark.py

To this end, we first wrote a test.py to provide our model demo. Code upload to https://github.com/gao3656888/SCI
